src/NuGet.Core/NuGet.Common/ConcurrencyUtilities.cs

```diff
@@ -265,5 +329,15 @@ private static string FilePathToLockName(string filePath)
                 return EncodingUtility.ToHex(hash, HashLength);
             }
         }
+
+        private static void EnterGlobalMutex()
+        {
+            GlobalMutex?.WaitOne();
```

Comment:
Should we wait until it is safe to enter or call other [overloaded methods of `WaitOne`](https://docs.microsoft.com/en-us/dotnet/api/system.threading.waithandle.waitone?view=net-5.0) which accepts `timeout` parameter?